syntax = "proto3";

option go_package = ".;ccanalyzer_client";

// 引入 google/protobuf/struct.proto
import "google/protobuf/struct.proto";

service CcAnalyzer {
  rpc AnalyzeFileContent (AnalyzeFileContentReq) returns (AnalyzeFileContentRes);
}

message AnalyzeFileContentReq {
  string language = 1;
  string file_content = 2;
}

message AnalyzeFileContentRes {
  // 使用 google.protobuf.Struct 表示动态键值对
  google.protobuf.Struct analyzed_data = 1;
}